enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,422 @@
+* **Current behavior:** If customer only want to include custom CRs from git directory to build site specific CRs:
+  * **inclusionDefault** must be set to *exclude* which will exclude all the CRs from ztp-container
+  * pass the list of CR file names under **filter.include**, that will include listed CRs. But it limits:
+
+    * does not allow to have the [same CR name](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go#L462-L464) for the custom CRs. 
+
+    * to be able to add custom CRs, customer needs to change the name of the custom CR.
+
+    * as a result, the user is not able to capture the full set of version-specific extra manifests in git along with the SiteConfig because they cannot override the default set because of the inability to have the same named file.
+
+  * below is an example of how to include a CR called `myCR.yaml` from `extraManifestPath` in current configurability: 
+
+
+  ```yaml
+  - cluster:
+    extraManifestPath: mypath/
+    extraManifests:
+      filter:
+        inclusionDefault: exclude
+        include:
+          - myCR.yaml					
+  ```
+
+
+### Goals
+
+* Customer can place the full set of version specific day-0 manifests (extra manifests) into git alongside the SiteConfig. This allows the user to deploy the correct version of manifests based on the version of OCP being deployed.
+
+* Allow customer to include multiple directory path to look for extra-manifests in siteconfig, this way we are better prepared for 4.15+ features
+
+* This will enable the customer to control what CRs get added in site generation
+
+* Keep backward compatibility of the behavior corresponds to 4.13 version, meaning the new parameter is not mandatory to generate CRs through siteconfig.
+
+### Non-Goals
+
+* Refactoring of the extra manifest filtering feature.
+### User Stories
+
+- [EPIC LINK](https://issues.redhat.com/browse/CNF-7365) --> [User Story](https://issues.redhat.com/browse/CNF-8672)
+
+## Proposal
+
+* The proposal is to use a new field as the opt-in, eg. `extraManifests.searchPaths` which can include multiple directories
+
+  ```bash
+  extraManifests:
+    searchPaths:
+      - myPath1/
+      - myPath2/
+  ```
+
+
+  * when `extraManifests.searchPaths` is defined in Siteconfig, the generated installation CRs will not include any builtin "extra manifest" CRs from the ztp-container.
+
+  * we will deprecate `extraManifestPath` in the next release. If defined it will continue to function exactly as it does today, but customers will be recommended to use the new extraManifests.searchPaths field.
+
+  * support multiple entries in the list in 4.14 – for example, recommended version specific extra-manifests in one directory and customer's custom extra manifests in another directory
+
+  * multiple entries accumulate. All files from entry1 and all files from entry2 will be applied during site generation.
```

Comment:
Properly should mention..if `fileA` exists in both entry1 and entry2...`entry2/fileA` provided that entry2 appear entryA? This is a change from pervious behaviour where fileA couldnt be included in extra-manifest folder if it existed in container? 

Or does this behaviour apply only when using this new feature?

Reply:
This behavior only applies when searchPaths is declared, later directory always get precedence. There is no change of behavior when searchPaths is not declared, meaning only applicable to new feature, i will reword that.

Reply:
Updated with commit ee93abc4c36ebfda332706059d8ba4c680dde7c1